Arduino-Bling/RobotBling/RobotBling.ino

```diff
@@ -0,0 +1,209 @@
+  SOME_BLING_STATE_3,
+
+  ANIMATION_COUNT // The number of animations
+};
+
+/*
+ * Magic delay function for driving animation library
+ * - Note that the delay() function mapped to _delay()
+ * - Anytime there is a delay() introduced to animations, it allows to check for new commands
+ * - commands drive the animation demo -- if a new command detected, we jump to the beginning
+ * of the loop -- see loop()
+ *
+ * important: _delay() is listening for printable characters '0' and
+ * above. This matches the message sent by robot's LED subsystem
+ */
+
+// A global variable to hold the current command to be executed
+uint8_t currentCommand = OFF;
+
+// The breadcrumb to get us back to the beginning of the loop
+jmp_buf env;
+
+// Our magic delay function
+void _delay(uint16_t timeout)
+{
+  while (timeout != 0)
+  {
+    delay(1);
+    timeout--;
+
+    // If there are any characters available on the serial port, read them
+    while (Serial.available() > 0)
+    {
+      char commandCharacter;
+      commandCharacter = Serial.read();
+      // use the ASCII table to identify the >= commandCharacter order
+      if (isprint(commandCharacter) && (commandCharacter >= '0'))
+      {
+        uint8_t command = commandCharacter - '0';
+        if (command < ANIMATION_COUNT)
+        {
+          // Found a valid command!
+          if (command != currentCommand)
+          {
+            // It's a new command!
+            currentCommand = command;
+            // Jump back to the top of the loop
+            longjmp(env, 1);
+          }
+        }
+      }
+    }
+  }
+}
+
+// redefining delay function
+#define delay _delay
+
+/*
+ * Long waited setup() -- clear the strip and set brightness level
+ */
+void setup()
+{
+  Serial.begin(9600); // required for listening to new commands
+
+  // Initialize all pixels to 'off' with default brightness
+  pixels.begin();
+  // set pixel brightness
+  setDefaultBrightness();
+  pixels.clear();
+  pixels.show();
+}
+
+/*
+ * loop() --
+ * - setup breadcrumb for magic _delay()
+ * - add animations to switch statement --> ensure matches the animation library above
+ */
+void loop()
+{
+  // Save a breadcrumb of where to jump back to
+  if (setjmp(env) != 0)
+  {
+    // If we jumped back (non-zero return value), then exit the loop()
+    // and it will be re-run again, leaving a new breadcrumb.
+    return;
+  }
+
+  // clear the strip between animation changes
+  pixels.clear();
+  setDefaultBrightness();
+  pixels.show();
+
+  // From here everything should be interruptable (to switch animations)
+  // as long as the animation code uses the `delay()` function.
+
+  // Take the current command and set the parameters, i.e. animations
+  switch (currentCommand)
+  {
+  // Annimations -- commands to match the enums above!
+  case OFF:
+    // all pixels are off
+    solid(rgbColor_OFF);
+    break;
+
+  case DISABLED:
+    // all pixels set to magenta
+    /* see solid() for use of helpers and basic animations */
```

Comment:
I don't think this comment is useful.
It might be helpful to rename the ```solid()``` function to something like ```display_solid_color()```

Reply:
will think about renaming - won't fix for now